libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/ActionButtonDescription.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.navigation.dropin
+
+import android.view.View
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+/**
+ * Custom Action Button description.
+ *
+ * @param view a view that will be added to the [NavigationView]
+ * @param position determines if the custom button should be placed before or after existing controls.
+ *                Defaults to Position.END.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class ActionButtonDescription(
+    val view: View,
+    val position: Position = Position.END
+) {
+    /**
+     * An enum that determines placement of the button.
+     */
+    enum class Position {
+        /**
+         * Place before existing controls.
+         */
+        START,
+
+        /**
+         * Place after existing controls.
+         */
+        END;
+    }
```

Comment:
I personally don't like using `Enum`. I would either replace this with `sealed class` or enumerated annotations

```
@ExperimentalPreviewMapboxNavigationAPI
class ActionButtonDescription(
    val view: View,
    @ButtonPositionDescriptor val position: String = END
) {
    companion object {
        const val START = "Start"
        const val END = "End"
    }

    @StringDef(START, END)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ButtonPositionDescriptor
}
```
Not a blocker though.